### DISCUSSION TITLE
How to use?

### DISCUSSION THREAD

**Discussion by zehawki** [khmyznikov/pwa-install#57] [Q&A] [answered]
@khmyznikov  The screens look very good, and certainly better than what I've already done for my apps. Perhaps I'm being dense, but for the life of me I cant figure out how to get this working for React. I took a look at the so called "react polyfill" - https://glitch.com/edit/#!/mango-shine-diagram?path=src%2Fcomponents%2Fpwa-install.jsx. It adds a button to click to start installation. But what would I do with that?

How does this drop-in automatically detect the browser & state and prompt accordingly?
There are a number of event listeners - who is issuing these events? What are these events anyway?
Inside the useEffect, `pwa-install-success-event` event handler is `handleInstallSuccess`, which calls `onInstallSuccess`, which itself is a trigger for the useEffect - didnt understand whats happening here.

Some detailed documentation would be nice.

**Comment by khmyznikov** [comment]
I'm not a React expert but you can reach the folks from related discussion here or in issues.

Yes the component detects everything itself, and in default mode you don't need a control. All events going from the component. You can change the behavior by component properties.

The main problem with React, it's a lack of native Web Component support, which should be fixed in React 19.

**Comment by khmyznikov** [comment, marked as answer]
The latest sample is here https://stackblitz.com/edit/vite-react-ts-2eeiak?file=src%2FApp.tsx
